src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -349,17 +355,42 @@ inline bool DnsStreamHandler::_filtering(DnsLayer &payload, [[maybe_unused]] Pac
             goto will_filter;
         }
         std::string_view qname_ci = payload.getFirstQuery()->getNameLower();
-        for (const auto &fqn : _f_qnames) {
-            // if it matched, we know we are not filtering
-            if (ends_with(qname_ci, fqn)) {
-                _static_suffix_size = fqn.size();
-                goto will_not_filter;
-            }
+        if (std::none_of(_f_qnames.begin(), _f_qnames.end(), [this, qname_ci](std::string fqn) {
+                if (ends_with(qname_ci, fqn)) {
+                    _static_suffix_size = fqn.size();
+                    return true;
+                }
+                return false;
+            })) {
+            // checked the whole list and none of them matched: filter
+            goto will_filter;
+        }
+    }
+    if (_f_enabled[Filters::GeoLocNotFound]) {
```

Comment:
That would already be achieved on a NetHandler right?

Reply:
@leoparente That was my thought as well. It's not ideal (from a user's perspective) that the metrics and filters associated with source IPs and ECS live in two different handlers, but I don't see an elegant way around that.

Reply:
@rboucher-ns1 but should we land it and then have a ticket to find a way to do that in an elegant way?

Reply:
I think these are two distinct filters, on two different handlers. I prefer them to stay separate so that a user can configure a policy to achieve his specific intent (i.e. whether to filter on notFound on source IP or ECS, or both).

What might be interesting in the future is chaining from NET handler to DNS handler, so that you could get DNS metrics on _only_ the traffic that has failed the GeoLoc or ASN lookup.